src/reviewRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -17,10 +17,148 @@
+ * @param {object} interaction
  */
-import template from 'taoQtiItem/reviewRenderer/tpl/interactions/textEntryInteraction';
-import textEntryInteraction from 'taoQtiItem/qtiCommonRenderer/renderers/interactions/TextEntryInteraction';
+const render = (interaction) => {
+    const attributes = interaction.getAttributes();
+
+    //setting up the width of the input field
+    if (attributes.expectedLength) {
+        const $input = interaction.getContainer();
+
+        //adding 2 chars to include reasonable padding size
+        const expectedLength = parseInt(attributes.expectedLength) + 2;
+        $input.css({
+            width: expectedLength + 'ch',
+            minWidth: expectedLength + 'ch'
+        });
+    }
+};
+
+const resetResponse = interaction => {
+    interaction.getContainer().text('');
+};
+
+/**
+ * Set the response to the rendered interaction.
+ *
+ * The response format follows the IMS PCI recommendation :
+ * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+ *
+ * Available base types are defined in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * Special value: the empty object value {} resets the interaction responses
+ *
+ * @param {object} interaction
+ * @param {object} response
+ */
+const setResponse = (interaction, response) => {
+    let responseValue;
+
+    try {
+        responseValue = pciResponse.unserialize(response, interaction);
+    } catch (e) {}
+
+    if (responseValue && responseValue.length) {
+        interaction.getContainer().text(responseValue[0]);
+    }
+};
+
+/**
+ * Return the response of the rendered interaction
+ *
+ * The response format follows the IMS PCI recommendation :
+ * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+ *
+ * Available base types are defined in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * @param {object} interaction
+ * @returns {object}
+ */
+const getResponse = interaction => {
+    const ret = { base: {} },
+        $input = interaction.getContainer(),
+        attributes = interaction.getAttributes(),
+        baseType = interaction.getResponseDeclaration().attr('baseType'),
+        numericBase = attributes.base || 10;
+    let value;
+
+    if ($input.hasClass('invalid') || (attributes.placeholderText && $input.text() === attributes.placeholderText)) {
+        //invalid response or response equals to the placeholder text are considered empty
+        value = '';
+    } else {
+        if (baseType === 'integer') {
+            value = locale.parseInt($input.text(), numericBase);
+        } else if (baseType === 'float') {
+            value = locale.parseFloat($input.text());
+        } else if (baseType === 'string') {
+            value = $input.text();
+        }
+    }
+
+    ret.base[baseType] = isNaN(value) && typeof value === 'number' ? '' : value;
```

Comment:
Number.isNaN

Reply:
I guess, yes since it's used in other places
https://github.com/oat-sa/tao-item-runner-qti-fe/blob/16d475c8ef51f8b03b3a3047dfcfab4adce0c831/src/qtiCommonRenderer/renderers/interactions/SliderInteraction.js#L230
https://github.com/oat-sa/tao-item-runner-qti-fe/blob/16d475c8ef51f8b03b3a3047dfcfab4adce0c831/src/reviewRenderer/renderers/interactions/ExtendedTextInteraction.js#L291

Reply:
There are different isNaN functions.

Number.isNaN checks exactly what you want to check while add another condition :
isNaN(value) && typeof value === 'number' -> Number.isNaN(value)